importccl,backupccl: create job record in planner txn

pbardea opened the pull request:
This commit updates the way that we create the job record in import and
applies the same logic to backup and restore.

Instead of committing the planner transaction before creating the job
record in a separate transaction, the job record creation now _uses_ the
planner transaction. The planner transaction is then committed after
creating the job record. This should not result in any change in
functionality. The change was intended for increased readability: since
the main goal of the plan hook is to create a job record, it makes sense
that it is done in the planner's transaction.

The same logic is applied to planning backup and restore jobs.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62379)
<!-- Reviewable:end -->

pbardea committed:
importccl,backupccl: create job record in planner txn

This commit updates the way that we create the job record in import and
applies the same logic to backup and restore.

Instead of committing the planner transaction before creating the job
record in a separate transaction, the job record creation now _uses_ the
planner transaction. The planner transaction is then committed after
creating the job record. This should not result in any change in
functionality. The change was intended for increased readability: since
the main goal of the plan hook is to create a job record, it makes sense
that it is done in the planner's transaction.

The same logic is applied to planning backup and restore jobs.

Release note: None

pbardea commented:
TFTR!
bors r+
